src/pages/patch/PatchRestartModal.tsx

```diff
@@ -71,10 +72,30 @@ export const PatchRestartModal: React.FC<PatchModalProps> = ({
   const patchBuildVariants = get(data, "patchBuildVariants");
   const [
     selectedTasks,
-    validStatus,
-    { toggleSelectedTask, setValidStatus },
+    patchStatusFilterTerm,
+    baseStatusFilterTerm,
+    { toggleSelectedTask, setPatchStatusFilterTerm, setBaseStatusFilterTerm },
   ] = usePatchStatusSelect(patchBuildVariants);
-
+  const { search } = useLocation();
+  const prevVisible = usePrevious(visible);
+  useEffect(() => {
+    // apply status filter from URL when the modal opens
+    if (visible && !prevVisible) {
```

Comment:
This is pretty cool idea to prevent unnecessary rerenders of the modal when its not in view.